### DISCUSSION TITLE
Explore: better auto-detection of log-data

### DISCUSSION THREAD

**Discussion by gabor** [grafana/grafana#44637] [Explore]
in explore, when data comes in, we detect what visualization to use.. we do it in two ways:
- if the dataframes have a preferred-visualization-type setting, we use that
- if not, we look at the shape of the data, and guess (for example, `isTimeSeries` here : https://github.com/grafana/grafana/blob/main/public/app/features/explore/utils/decorators.ts#L177-L182 )

there are cases where log-data comes in that does not have the preferred-visualization-type field set. for example, InfluxDB Flux queries that return timestamp+text data. (NOTE: a Flux query can return basically any data, metric, logs, etc.)

for these cases it would be great if explore could better autodetect this.

a couple potential approaches:
- similar to `isTimeSeries` add an `isLogs` function
- currently we have the visualization-suggestions in dashboards... we could maybe use that logic to choose the visualization in explore?
- add a way where the user can pick the visualization (logs/metrics/etc.) for every query in explore

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
